akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -578,7 +579,20 @@ private[akka] final class PromiseActorRef private (
       val promiseResult = message match {
         case Status.Success(r) => Success(r)
         case Status.Failure(f) => Failure(f)
-        case other             => Success(other)
+        case f: FaultyResponseMarker =>
+          f match {
+            case a: AlwaysFailingResponse => Failure(new RuntimeException(a.failureDescription))
+            case s: scaladsl.MaybeFailingResponse =>
+              s.failureDescription match {
+                case None                     => Success(s)
+                case Some(failureDescription) => Failure(new RuntimeException(failureDescription))
+              }
+            case j: javadsl.MaybeFailingResponse =>
+              val failureDescription = j.failureDescription
+              if (failureDescription.isEmpty) Success(j)
+              else Failure(new RuntimeException(failureDescription.get()))
+          }
+        case other => Success(other)
```

Comment:
why two levels of matching? It reads better without nesting.

Reply:
To avoid matching on three different types for every message in the happy case of not using this feature at all